Tools/AP_Periph/can.cpp

```diff
@@ -1297,6 +1301,17 @@ void AP_Periph_FW::can_battery_update(void)
         fix_float16(pkt.current);
         fix_float16(pkt.temperature);
 
+        pkt.state_of_health_pct = UAVCAN_EQUIPMENT_POWER_BATTERYINFO_STATE_OF_HEALTH_UNKNOWN;
+        pkt.state_of_charge_pct = battery.lib.capacity_remaining_pct(i);
+        pkt.model_instance = i+1;
+
+        // example model_name: "org.ardupilot.ap_periph SN 123"
+        char text[UAVCAN_EQUIPMENT_POWER_BATTERYINFO_MODEL_NAME_MAX_LENGTH+1];
+        pkt.model_name.len = hal.util->snprintf(
```

Comment:
snprintf returns size you would need to hold whole string, so need to use strlen

Reply:
changed to use strlen()